docs/source/developer-docs/index.rst

```diff
@@ -16,3 +16,17 @@ Developer Guide
 Whether you are a new contributor or a seasoned developer, we're pleased that
 you are working on Jupyter. We hope you find the Developer Guide is useful.
 Please suggest changes or ask questions about the contents. Thanks!
+
+If you are interested in installing a specific project from source, each
+project has documentation on ReadTheDocs. For example, IPython documentation
+can be found on `ReadTheDocs <http://ipython.readthedocs.io/en/latest/install/install.html#installation-from-source>`_.
+Most of our packages can be installed from the source directory like any other
+Python package, by running:
+
+::
+
+    pip install .
+
+The Jupyter notebook needs some extra pieces to build Javascript components;
+the information about that is available on ReadTheDocs
+`here <http://jupyter-notebook.readthedocs.io/en/latest/contributing.html#setting-up-a-development-environment>`_.
```

Comment:
Minor pet peeve: I don't like links saying 'here', because if you just glance at the link it means nothing. (Of course, I still do it on occasion ;-)

I'd rather see something like:

> ... the information about that is __in the notebook contributor documentation__.